core/src/main/kotlin/com/exactpro/th2/read/db/impl/grpc/DataBaseReaderGrpcServer.kt

```diff
@@ -168,33 +170,59 @@ class DataBaseReaderGrpcServer(
     }
 
     private class GrpcExecuteListener(
-        private val observer: StreamObserver<QueryResponse>,
+        streamObserver: StreamObserver<QueryResponse>,
+        private val executionId: Long,
         private val event: Event,
         private val onEvent: (Event) -> Unit,
     ) : ResultListener {
+        private val skipped = AtomicLong()
+        private val observer = streamObserver as ServerCallStreamObserver<QueryResponse>
+
         override fun onRow(sourceId: DataSourceId, row: TableRow) {
-            requireNotNull(row.executionId) {
-                "'Execution id' is null for row: $row"
+            check(executionId == row.executionId) {
+                "'Execution id' isn't equal to '$executionId', row: $row"
+            }
+
+            while (!observer.isReady) {
+                if (observer.isCancelled) {
+                    skipped.incrementAndGet()
+                    return
+                }
+                Thread.yield()
+            }
```

Comment:
note: This is not the right approach to implement backpressure when we use coroutines, in my opinion. I think that we need to do is to make `onRow` suspendable, create a channel with 1 capacity, implement something like [this](https://stackoverflow.com/a/55421973) and call resume method in `setOnReadyHandler` runnable

```
if (!observer.isReady) {
  channel.receive()
}
```
In the `setOnReadyHandler` we can call `channel.trySend`